docs(privatelink): rename to Private Links and update for self-serve UI (#77830)
airbytehq/airbyte: docs/platform/operating-airbyte/privatelink.md

---
products: cloud-teams

> sidebar_label: Private Links

Reviewer: Devin remove this line and use the H1 as the sidebar label.

---

# Private Links

Private Links allow Airbyte Cloud to securely connect to your data sources and destinations without using the public internet or exposing IPs. Traffic between Airbyte and your resources stays entirely within your cloud provider's private network, providing enhanced security for organizations with strict network isolation requirements.

:::info

> Private Links is an add-on feature for Airbyte Cloud. To enable Private Links for your organization, [talk to sales](https://airbyte.com/company/talk-to-sales).

Reviewer: ```suggestion
Private Links is an add-on feature for Airbyte Cloud, and requires the Pro plan or above. To enable Private Links for your organization, [talk to sales](https://airbyte.com/company/talk-to-sales).
```

### Single tenant or multi tenant

Airbyte Cloud is multi-tenant cloud software, regardless of whether you use Private Links. However, Private Links can feel like a single-tenant experience due to the lack of a shared public endpoint and strong isolation at your network boundary. With or without Private Links, all sync jobs in Airbyte use separate Kubernetes pods. This pod-level isolation further helps ensure data syncs from one organization are never exposed to another organization. Your syncs never run in their pods, and their syncs never run in your pods.

### Availability and supported regions

Private Links is currently available for AWS. If you need Private Links for Azure or GCP, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.

| Cloud provider | Name                            | Available                                                  |
| -------------- | ------------------------------- | ---------------------------------------------------------- |
| AWS            | PrivateLink                     | Yes                                                        |

> | Azure          | Private Endpoint / Private Link | [Talk to sales](https://airbyte.com/company/talk-to-sales) |

Reviewer 2: We do not support Azure? Not sure where this came from? We have 2 AWS Dataplanes in `us-east-1` and `eu-west-3`. Both support cross-region endpoints.

Reviewer 2: Setting up Azure will not be trivial. I don't think we should be selling this yet.

Reviewer 3: Talked to @katmarkham about this. I don't think the intent here is to sell it right now but more to get a sense of how many times we get a request for it.
But by getting people to talk to sales we can get more data and plan accordingly / decide if this is worth adding to the roadmap.
We should keep it the way it is right now

| GCP            | Private Service Connect         | [Talk to sales](https://airbyte.com/company/talk-to-sales) |

Airbyte maintains AWS PrivateLink infrastructure in the US (`us-east-1`) and EU (`eu-west-3`). Your service can be in any AWS region. Airbyte uses [cross-region PrivateLink](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-cross-region-connectivity-for-aws-privatelink/) to connect to services outside of these regions, so you don't need to relocate your infrastructure.

### Cross-region connectivity

Airbyte's PrivateLink endpoints are located in `us-east-1` (US) and `eu-west-3` (EU). If your VPC endpoint service is in a different AWS region, AWS cross-region PrivateLink transparently bridges the connection. For this to work, you must enable cross-region support on your VPC endpoint service and add Airbyte's region to your service's supported regions list. See [Cross-Region Connectivity for AWS PrivateLink](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-cross-region-connectivity-for-aws-privatelink/) for details.

### Limitations

> Private Links are currently limited to sources and destinations running in AWS. This includes services like Snowflake, Databricks, and PostgreSQL hosted on AWS. If you need to connect to resources outside of AWS, Private Links are not yet available.

Reviewer 2: They ONLY support NLB backend endpoint services.

### Single tenant or multi tenant

Airbyte Cloud is multi-tenant cloud software, regardless of whether you use Private Links. However, Private Links can feel like a single-tenant experience due to the lack of a shared public endpoint and strong isolation at your network boundary. With or without Private Links, all sync jobs in Airbyte use separate Kubernetes pods. This pod-level isolation further helps ensure data syncs from one organization are never exposed to another organization. Your syncs never run in their pods, and their syncs never run in your pods.

### Availability and supported regions

Private Links is currently available for AWS only. Airbyte doesn't support Azure Private Endpoint or Google Cloud Private Service Connect.

| Cloud provider | Name                            | Available |
| -------------- | ------------------------------- | --------- |
| AWS            | PrivateLink                     | Yes       |

> | Azure          | Private Endpoint / Private Link | No        |

Author: ```suggestion
| Azure          | Private Endpoint / Private Link | [talk to sales](https://airbyte.com/company/talk-to-sales)        |
```

> | GCP            | Private Service Connect         | No        |

Author: ```suggestion
| GCP            | Private Service Connect         | [talk to sales](https://airbyte.com/company/talk-to-sales)        |
```

Airbyte maintains AWS PrivateLink infrastructure in the US (`us-east-1`) and EU (`eu-west-3`). Your service can be in any AWS region. Airbyte uses [cross-region PrivateLink](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-cross-region-connectivity-for-aws-privatelink/) to connect to services outside of these regions, so you don't need to relocate your infrastructure.

### Cross-region connectivity

Airbyte's PrivateLink endpoints are located in `us-east-1` (US) and `eu-west-3` (EU). If your VPC endpoint service is in a different AWS region, AWS cross-region PrivateLink transparently bridges the connection. For this to work, you must enable cross-region support on your VPC endpoint service and add Airbyte's region to your service's supported regions list. See [Cross-Region Connectivity for AWS PrivateLink](https://aws.amazon.com/blogs/networking-and-content-delivery/introducing-cross-region-connectivity-for-aws-privatelink/) for details.

### Limitations

> Private Links are currently limited to sources and destinations running in AWS. This includes services like Snowflake, Databricks, and PostgreSQL hosted on AWS. If you need to connect to resources outside of AWS, Private Links are not an option.

Author: ```suggestion
Private Links are currently limited to sources and destinations running in AWS. This includes services like Snowflake, Databricks, and PostgreSQL hosted on AWS. If you need to connect to resources outside of AWS, Private Links are not yet available.
```

> Support for AWS S3 sources and destinations is in development. If you need a Private Link for an S3 bucket, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.

Author: ```suggestion
Support for AWS S3 buckets is in development. If you need a PrivateLink for an S3 bucket, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.
```

> Support for AWS S3 buckets is in development. If you need a PrivateLink for an S3 bucket, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.

Reviewer: ```suggestion
If you need a PrivateLink for an S3 bucket, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.
```

Reviewer: We've burned ourselves with future promises before.

For managed AWS services like RDS or Aurora, additional configuration is required to expose them via AWS PrivateLink. See [Using Private Links with managed services](#using-private-links-with-managed-services) for details.

## Prerequisites

> Before setting up a Private Link, ensure you have the following:

Reviewer: The terminology waters are getting muddied here. Because we only support AWS, these steps specifically describe PrivateLink, the AWS brand, and don't need to use generic language.

Reviewer: Devin ignore this comment.

> Before setting up a Private Link, ensure you have the following:

Reviewer: ```suggestion
Before setting up PrivateLink, ensure you have the following:
```

Private Links are currently limited to sources and destinations running in AWS. This includes services like Snowflake, Databricks, and PostgreSQL hosted on AWS. If you need to connect to resources outside of AWS, Private Links are not yet available.

If you need a PrivateLink for an S3 bucket, [talk to sales](https://airbyte.com/company/talk-to-sales) so we can scope your requirements.

For managed AWS services like RDS or Aurora, additional configuration is required to expose them via AWS PrivateLink. See [Using Private Links with AWS managed services](#using-private-links-with-aws-managed-services) for details.

## Prerequisites

> Before setting up PrivateLink, ensure you have the following:

Reviewer: I think the previous state for line 71, "a Private Link" was correct.

  - `elasticloadbalancing:CreateTargetGroup`
  - `elasticloadbalancing:RegisterTargets`

> ## Set up PrivateLink

Reviewer: The previous state for line 89 was correct, "a Private Link".

  - `elasticloadbalancing:CreateTargetGroup`
  - `elasticloadbalancing:RegisterTargets`

> ## Set up a Private Link

Reviewer: ```suggestion
## Set up PrivateLink
```

4. Note your endpoint service name. It follows the format `com.amazonaws.vpce.{region}.vpce-svc-{id}`. You provide this to Airbyte in Step 4.

### Step 3: Configure permissions for Airbyte

Add Airbyte's AWS account as an allowed principal on your endpoint service. This allows Airbyte to create a VPC endpoint that connects to your service. Airbyte provides this account number for you.

See [Manage permissions](https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#add-remove-permissions) in the AWS documentation for instructions on adding principals to your endpoint service.

### Step 4: Create the Private Link in Airbyte

Create the Private Link from your Airbyte Cloud workspace settings.

1. In Airbyte Cloud, go to **Workspace settings** > **Private Links**.
2. Enter a **Name** for the Private Link (for example, `my-production-link`).
3. Enter your **Endpoint Service Name** (for example, `com.amazonaws.vpce.us-east-1.vpce-svc-05df72d283c05fbc7`).
4. Click **Create Private Link**.

Airbyte creates a VPC endpoint that connects to your endpoint service.

> If your data source or destination requires additional DNS configuration (for example, Snowflake), share the service type and any required configuration with Airbyte after creating the Private Link. See [Using Private Links with Snowflake](#using-private-links-with-snowflake) for details.

Reviewer 2: We do not currently support snowfalke endpoints through automation. It requires additional provisioning.

Author: @bernielomax I don't think this contradicts that.  There's a whole section on snowflake 
<img width="904" height="230" alt="Screenshot 2026-05-08 at 7 51 39 AM" src="https://github.com/user-attachments/assets/bb127c27-b6f5-412c-8776-c6f5798e1798" />

Reviewer 2: Quick clarification — for Snowflake specifically, are customers expected to run Step 4 (self-serve through UI) themselves and then raise a ticket for the DNS, or should the whole process go through support? The doc reads ambiguously between those today, and if it isn't clear to me I suspect it won't be clear to a customer either.

Author: @bernielomax I think we want them to do step 4 and then reach out. The more of the process we can have them do, the better

### Step 5: Accept the connection request

After Airbyte creates the VPC endpoint, the connection appears as **Pending Acceptance** in your AWS console under **VPC > Endpoint Services > Endpoint Connections**. You must accept this connection request before the Private Link becomes active.

Once you accept the request, the connection status changes to **Available** and your Private Link is ready to use with your Airbyte connections.

## Using Private Links with AWS managed services

> Managed AWS services like Amazon RDS and Aurora don't natively support VPC endpoint services. To use Private Links with these services, you need to set up additional infrastructure to expose them via a Network Load Balancer.

Reviewer 2: They MUST be behind the customers NLB. We do not host these services for customers.

3. If your service is in a different region than Airbyte's PrivateLink infrastructure (`us-east-1` for US, `eu-west-3` for EU), enable cross-region support on your VPC endpoint service and add Airbyte's region to the supported regions list. See [Cross-region connectivity](#cross-region-connectivity) for more information.

> 4. Note your endpoint service name. It follows the format `com.amazonaws.vpce.{region}.vpce-svc-{id}`. You'll provide this to Airbyte in Step 4.

Reviewer: Devin: this line should have immediately failed linting due to future tense. Can you confirm ReviewDog ran and suppressed commenting due to its config? Why did you not lint locally?

Reviewer: Continue suppressing Vale messages in CI but lint with Vale locally always and try to comply.

Create the Private Link from your Airbyte Cloud workspace settings.

1. In Airbyte Cloud, go to **Workspace settings** > **Private Links**.
2. Enter a **Name** for the Private Link (for example, `my-production-link`).
3. Enter your **Endpoint Service Name** (for example, `com.amazonaws.vpce.us-east-1.vpce-svc-05df72d283c05fbc7`).
4. Click **Create Private Link**.

Airbyte creates a VPC endpoint that connects to your endpoint service.

If your data source or destination requires additional DNS configuration (for example, Snowflake), share the service type and any required configuration with Airbyte after creating the Private Link. See [Using Private Links with Snowflake](#using-private-links-with-snowflake) for details.

### Step 5: Accept the connection request

After Airbyte creates the VPC endpoint, the connection appears as **Pending Acceptance** in your AWS console under **VPC > Endpoint Services > Endpoint Connections**. You must accept this connection request before the Private Link becomes active.

Once you accept the request, the connection status changes to **Available** and your Private Link is ready to use with your Airbyte connections.

> ## Using Private Links with managed services

Reviewer: ```suggestion
## Using Private Links with AWS managed services
```
